protoc_plugin/test/import_public_test.dart

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env dart
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import 'package:test/test.dart';
+
+import '../out/protos/import_public.pb.dart';
+
+main() {
+  test('can reference a message type imported publicly', () {
+    expect(Foo(), TypeMatcher<Foo>());
```

Comment:
nit: a slightly more fluent way to use `TypeMatcher` is to use `isA<Foo>()` instead of the matcher directly: https://api.flutter.dev/flutter/package-matcher_matcher/isA.html

Reply:
I used isA at first, but that is incompatible with the versions of test that support the oldest sdk we support.
Maybe we should soon raise the lower sdk constraint?